### ISSUE TITLE
Handling SIGHUP

### ISSUE THREAD

**Issue by colmonty** [kanongil/exiting#3]
Hi,

Firstly, thanks for writing `Exiting` - it's allowed my HapiJS docker containers to exit quickly and cleanly, as opposed to having to wait 10s to be killed.

When I attempt to send a `SIGHUP` to my docker container (using any of the methods [here](https://blog.confirm.ch/sending-signals-docker-container/)), my HapiJS container shuts down and reports:

```text
[exiting] Server stop failed: Error: Process aborted
```

I need to handle `SIGHUP` to reload secrets every n minutes, when they are rotated by Hashicorp Vault and other parts of the infra send a `SIGHUP` to each container which cares about this.

The basics of my server:

```javascript
const composer = require('./index');
const Exiting = require('exiting');

composer((err, server) => {

  if (err) {
    throw err;
  }

  new Exiting.Manager(server).start(() => {
    console.log(`Started Service on port ${server.info.port}`);
  });

  server.on('stop', () => {
    console.log('Server stopped');
  });

  process.on('SIGHUP', () => {
    console.log(`SIGHUP received`);
  });
});
```

Any ideas what might be going on? I'm not seeing either `Server stopped` or `SIGHUP received` being logged.

**Comment by colmonty** [comment]
I was noting in `index.js` within the package that `SIGHUP` is declared but set to `false` - I'm not understanding what it's doing, but it does _kind of_ suggest it's not being handled?

```javascript
const internals = {
    manager: null,
    signals: {
        SIGINT: true,
        SIGQUIT: true,
        SIGTERM: true,
        SIGHUP: false
    },
    processExit: null
};
```

**Comment by colmonty** [comment]
when I remove `Exiting` from the equation, `SIGHUP` is handled - so I _could_ go that route, and maybe find another package which will handle the exit signals and also allow me to handle `SIGHUP`. But I'd rather help out and get this working with `Exiting`.

**Comment by kanongil** [comment]
`SIGHUP` is treated as an abnormal exit condition, which normally happens when you close the terminal window, where you have started the server.

You should not send this signal to the server, but rather directly to the vault process. If you need to relay the signal through the server, you should probably use the `SIGUSR1` signal instead.

**Comment by colmonty** [comment]
thanks for the reply @kanongil 

Let's ignore the Vault bit - that's just confusing matters, sorry. I'd never send it a signal as it just stores secrets. Anyway, this is all controlled by the infrastructure and orchestration which I deploy into. I have no control over this. 

Currently, Exiting is incompatible for me because it treats `SIGHUP` as an abnormal exit condition, as you state.

The NodeJS docs tell me that this is a somewhat confusing signal, but my takeaway from [this](https://nodejs.org/api/process.html#process_signal_events) doc is that you're taking the Windows
stance?

Any chance of a config switch so users of Exiting can decide which way they want it to be handled?

**Comment by colmonty** [comment]
@kanongil I've created a PR for you: https://github.com/kanongil/exiting/pull/4

However, the CI tests are failing on recent Node versions due to:
`The following leaks were detected:WebAssembly`

For some reason my terminal tab was running a really old version of Node initially, so I didn't catch it. I can replicate the error locally, but I can't for the life of me find what's going on or how to fix it. Any ideas?

**Comment by colmonty** [comment]
actually, reverted to your master and getting the same error on `npm test`

**Comment by colmonty** [comment]
I've closed PR#4 as it simply wasn't doing what I thought.

New PR created: https://github.com/kanongil/exiting/pull/5
This one actually works.

Could you please take a look and consider merging it @kanongil ?

Currently I'm having to install my forked version directly from github, which is far from ideal (for one, I've had to add git to my docker image).
Having a forked version of exiting in the npm registry doesn't help anyone, I don't think, so I'd rather not follow that route.

(still getting the Memory Leak for WebAssembly, but that exists on master as well, as I said above)
